### DISCUSSION TITLE
[ScreenSelectMusic] chart preview per player

### DISCUSSION THREAD

**Discussion by Horsey-** [TeamRizu/OutFox#261] [03 - Feature Requests]
![44262473_10157695542293765_3741278864026894336_o](https://user-images.githubusercontent.com/5679966/84597538-616a6700-ae19-11ea-93a7-c45a2f25bb4a.jpg)

Would be great to have a 1st party module to display a live chart preview. The above image is a concept, not a currently working prototype in SM5.1 . 

Is there a chart parser implemented as a module in OutFox (yet)? I'm not experienced enough to completely code this myself, but I can absolutely help with implementation and theme-side assets. 

Specifics:
-play chart preview in sync with song preview
-arrows display as if on autoplay (holds/rolls/lifts trigger properly)
-show where in the song by displaying an indicator on the NPS graph (a white line, at the exact moment in the preview atop a greyed portion from the beginning to the end of the song preview?)
-NPS graph scrolls/zooms in the case of long songs
-respects noteskin changes
-respects direction changes ("reverse"?)
-respects note modifiers? (dizzy, mini, flipped, no-mines, no jumps, etc)

**Comment by moruzerinho6** [comment]
Hello @Horsey- 

Project Moondance Discord Moderator here, a while ago we created a bug hunter system, users who make issue on this repository gain points and can be rewarded in our discord, so I'm here to ask, are you in Project Moondance discord?

If not, you can join using this link: https://discord.gg/3DgsbSg

If you don't wish to join, that's okay. Please be sure to contact any moderator if you join.

**Comment by Horsey-** [comment]
contact a moderator about this feature request, or do I need to be verified on the discord server?

**Comment by moruzerinho6** [comment]
If you have a account in the discord, just enter our server that I linked in my reply, and mention a moderator that it's you from this issue.
